### PR TITLE
Winningestworst coach

### LINKED ISSUE #25: Winningest/Worst Coach

### PR DESCRIPTION
**Pull request by pcbennett108**
## Describe your changes in detail
Add methods and tests for winningest_coach & worst_coach.

## Checklist before requesting a review
- [X ] I have performed a self-review of my code
- [ X] I have added thorough tests
- [X ] I AM MERGING WITH Westonio/futbol

## Anything else noteworthy? (E.g. note anything that might need refactoring or special attention later)

1) These methods are also due for a refactor. I wanted to wait for the other refactor branch to merge before pinning this branch's refactors to it. It may well be possible but I am still unsure of git behavior in that regard.

2) There is a tie for coach with most wins in at least one season. This MAY cause issue with spec harness. There may be a tie for worst in other seasons. I may have to put in logic to handle that but I would like to see what spec harness says first.

3) The iteration parameters for these methods refer to "...win percentage for the season". I think this is an instance or wording clarity. I only gathered data for most/fewest wins. A percentage would be unnecessary if we are only comparing coaches in their respective seasons as the season length is the same for all. Plus, the required return is just the coach's name.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -127,12 +127,33 @@ def lowest_scoring_home_team
...
 
-  def worst_coach
-
+  def worst_coach(season_id)
+    games_in_season = @game_teams.find_all do |game|
+      game.game_id[0..3] == season_id[0..3]
+    end
+    grouped_by_coach = games_in_season.group_by { |game| game.head_coach }
+    wins_by_coach = grouped_by_coach.each do |coach, games|
+      grouped_by_coach[coach] = games.find_all { |game| game.result == "WIN"}.length
```
Comment (topher-nullset) [inline lib/stat_tracker.rb:152]: Really nice logic here

**Review by topher-nullset — APPROVED** [review]
I like it!

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -127,12 +127,33 @@ def lowest_scoring_home_team
 
   # Season Statistics
 
-  def winningest_coach
-
+  def winningest_coach(season_id)
+    games_in_season = @game_teams.find_all do |game|
+      game.game_id[0..3] == season_id[0..3]
+    end
```
Comment (westonio) [inline lib/stat_tracker.rb:133]: This is just a note that we'll want to update this portion to match the helper method created on another branch -> .season_games(season_id)

**Review by westonio — APPROVED** [review]
Looks good @pcbennett108 ! Thanks for finishing that out 🥇

**Pull request merged**